error: 'unreachable' macro redefined

https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/util.h#L46

clang has added a definition to unreachable.

```
In file included from external/jemalloc_new/include/jemalloc/internal/assert.h:2:
external/jemalloc_new/include/jemalloc/internal/util.h:43:9: error: 'unreachable' macro redefined [-Werror,-Wmacro-redefined]
   43 | #define unreachable() JEMALLOC_INTERNAL_UNREACHABLE()
      |         ^
prebuilts/clang/host/linux-x86/clang-r530567/lib/clang/19/include/__stddef_unreachable.h:16:9: note: previous definition is here
   16 | #define unreachable() __builtin_unreachable()
      |         ^
```

--- a/include/jemalloc/internal/util.h
+++ b/include/jemalloc/internal/util.h
@@ -39,12 +39,6 @@
 #  define unlikely(x) !!(x)
 #endif
 
-#if !defined(JEMALLOC_INTERNAL_UNREACHABLE)
-#  error JEMALLOC_INTERNAL_UNREACHABLE should have been defined by configure
-#endif
-
-#define unreachable() JEMALLOC_INTERNAL_UNREACHABLE()
-
 /* Set error code. */
 UTIL_INLINE void
 set_errno(int errnum) {

--- a/include/jemalloc/internal/util.h
+++ b/include/jemalloc/internal/util.h
@@ -39,6 +39,15 @@
 #  define unlikely(x) !!(x)
 #endif
 
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L
+#include <stddef.h>
+#else
+#if !defined(JEMALLOC_INTERNAL_UNREACHABLE)
+#  error JEMALLOC_INTERNAL_UNREACHABLE should have been defined by configure
+#endif
+#define unreachable() JEMALLOC_INTERNAL_UNREACHABLE()
+#endif
+
 /* Set error code. */
 UTIL_INLINE void
 set_errno(int errnum) {

